core/src/main/clojure/xtdb/xtql.clj

```diff
@@ -685,47 +710,52 @@
      {:ra-plan
       [:unnest {unnested-col pre-col}
        [:map [{pre-col expr}]
-        (wrap-expr-subqueries acc-plan subqueries)]]}
+        (wrap-expr-subqueries acc-plan acc-provided-vars subqueries)]]}
      (update var->cols original-unnested-col (fnil conj #{}) unnested-col))))
 
 (defn wrap-unnests [acc-plan unnests]
   (reduce wrap-unnest acc-plan unnests))
 
-(defn wrap-inner-join [acc-plan {:keys [args] :as join-plan}]
-  (if (seq args)
+(defn wrap-inner-join [{acc-provided-vars :provided-vars :as acc-plan}
+                       {:keys [arg-subqueries tmp-expr-sym->apply-param-sym tmp-expr-sym->expr-vec] :as join-plan}]
+  (if (seq tmp-expr-sym->expr-vec)
     (let [{:keys [rels var->cols]} (with-unique-cols [acc-plan join-plan])
           [{acc-plan-with-unique-cols :ra-plan}
            {join-subquery-plan-with-unique-cols :ra-plan}] rels]
       (wrap-unify
-       {:ra-plan [:apply :cross-join args
-                  acc-plan-with-unique-cols
+       {:ra-plan [:apply :cross-join tmp-expr-sym->apply-param-sym
+                  [:map tmp-expr-sym->expr-vec
+                   (wrap-expr-subqueries acc-plan-with-unique-cols acc-provided-vars arg-subqueries)]
```

Comment:
I am implicitly making usage of the fact that `with-unique-cols` doesn't change the column names of the first plan given (`acc-plan` in this case). Hence I don't need to cleanup the `tmp-expr-sym`. I don't know if this approach is bad.

Reply:
IIRC this code was already making use of that assumption, so even if it is a mistake its not one introduced here.

When you say you don't need to clean up, you mean explicitly right? In that it will be projected away implicitly as a result of calling `wrap-unify` with the original col names.

Reply:
Yes exactly.